server/service/cron_schedules.go

```diff
@@ -0,0 +1,15 @@
+package service
+
+import (
+	"context"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+// TriggerCronSchedule attempts to trigger an ad-hoc run of the named cron schedule.
+func (svc *Service) TriggerCronSchedule(ctx context.Context, name string) error {
+	if err := svc.authz.Authorize(ctx, &fleet.CronSchedules{}, fleet.ActionWrite); err != nil {
```

Comment:
We should add a test to make sure this works only for the users/roles we want. There are other tests you can mimic for this.